pkg/controllers/provisioningv2/rke2/machineprovision/controller.go

```diff
@@ -193,7 +193,8 @@ func (h *handler) getMachineStatus(job *batchv1.Job) (rkev1.RKEMachineStatus, er
 	if job.Spec.Template.Labels[InfraJobRemove] == "true" {
 		condType = deleteJobConditionType
 	}
-	if !job.Status.CompletionTime.IsZero() {
```

Comment:
@Oats87 If we reconcile the job deletion even though it was successful, it would reset the `Ready` condition to false and the infrastructure would never be considered Ready by CAPI.

Reply:
this is the kind of stuff I worry about with us messing with all of these conditions.

Reply:
Me too, FWIW I used to be able to reproduce this intermittently on 2.6, so we never fixed this issue, its still just waiting to happen in 2.6 release line.